### PR TITLE
Make top-level filters optional

### PR DESCRIPTION
**Pull request by as2388**

### CODE REVIEW

File: src/pipeline_configuration_spec.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import json
 from dataclasses import dataclass
+from typing import Optional
```
Comment (IsaackMwenda) [inline src/pipeline_configuration_spec.py:3]: remove unused imports

File: src/engagement_db_to_analysis/traced_data_filters.py [hunk 2]
```diff
@@ -42,14 +43,17 @@ def rqa_time_range_filter(user, messages_traced_data, pipeline_config):
     filtered = []
     for td in messages_traced_data:
         if td["dataset"] in rqa_engagement_db_datasets:
-            if start_time_inclusive <= isoparse(td["timestamp"]) < end_time_inclusive:
-                td.append_data(td, Metadata(user, Metadata.get_call_location(), time.time()))
-                filtered.append(td)
+            if start_time_inclusive is not None and isoparse(td["timestamp"]) < start_time_inclusive:
```
Comment (IsaackMwenda) [inline src/engagement_db_to_analysis/traced_data_filters.py:46]: Its great that we can filter with either of them independently.

File: src/engagement_db_to_analysis/traced_data_filters.py [hunk 3]
```diff
@@ -42,14 +43,17 @@ def rqa_time_range_filter(user, messages_traced_data, pipeline_config):
...
+                continue
+            if end_time_inclusive is not None and isoparse(td["timestamp"]) > end_time_inclusive:
+                continue
+            td.append_data(td, Metadata(user, Metadata.get_call_location(), TimeUtils.utc_now_as_iso_string()))
+            filtered.append(td)
         else:
             filtered.append(td)
 
     log.info(f"Filtered out messages sent outside the time range "
```
Comment (IsaackMwenda) [inline src/engagement_db_to_analysis/traced_data_filters.py:55]: update to log appropriately when one of them is None.

**Review by IsaackMwenda — APPROVED** [review]

File: src/pipeline_configuration_spec.py (hunk 1 above)
Comment (as2388) [reply src/pipeline_configuration_spec.py]: fixed

File: src/engagement_db_to_analysis/traced_data_filters.py (hunk 3 above)
Comment (as2388) [reply src/engagement_db_to_analysis/traced_data_filters.py]: done

**Pull request merged**